### PR TITLE
feat: make header dynamic in trade modal

### LINKED ISSUE #711: UI: Dynamic header w/ back button in trade modal
There should be a dynamic header in the trade modal that can be changed from screen to screen.  There should also be a back button to go to a previous screen (from within the modal).

The "Trade" Header in the first screenshot should be removed an replaced with a Dynamic header with the "Confirm Trade" and back button"

This Back button should be present when selecting the input/output assets as well. 

Change the modal header to "Confirm Details" on confirm screen

Move the block header into each screen, so that we can have different things on each screen.

Screen Shot 2022-01-10 at 2 03 03 PM

Screen Shot 2022-01-10 at 2 03 56 PM

Screen Shot 2022-01-10 at 2 02 40 PM

### PR DESCRIPTION
**Pull request by helr441**
## Description

Makes the header in Trade modal dynamic, so that individual screens can customize it. Below are the Trade screens and their corresponding header titles and buttons (if any):

1. `TradeInput` - "Trade" (no buttons)
2. `TradeConfirm` - "Confirm Details" (with back button)
3. `TradeApproval` - "Trade" (no buttons)
4. `SelectAsset` - "Trade" (with back button)

The approach taken was to drop the `Card` tag (as well as `Card.Header` and `Card.Body`) that surrounded the different Trade screens in `TradeCard`, and have the screens have their own `Card` tag (as well as `Card.Header` and `Card.Body`).

This PR also includes some clean ups encountered while we were at it, such as removing `CardActions`, a duplicate of `TradeCard`.

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [x] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (~Breaking~/Non-breaking Change)

## Issue (if applicable)

Closes #711.

## Testing

Please outline all testing steps

1. Pull branch locally and run `yarn` to install new deps
2. Run `yarn dev`
3. In Trade modal on right of Dashboard page, check that you can see the TradeInput screen with header "Trade". There should be only one header "Trade", instead of one header "Trade" and one more "Trade" header below.
4. Clicking the token (for both buy or sell) brings you to the SelectAsset screen with header "Trade" and a back button. Clicking the back button brings you back to TradeInput screen.
5. Entering an amount and clicking "Preview Trade" brings you to TradeConfirm screen with header "Confirm Details", instead of "Confirm Trade".
6. TBD Approval screen (locally I basically modified the route navigated to from `TradeConfirm` to `Approval` in TradeInput....)

## Screenshots (if applicable)

Locally I modified the dashboard page to display the old Trade modal (left) and the one in this PR (right). Here are the screenshots:

1. TradeInput: TradeInput
2. SelectAsset: SelectAsset
3. Approval: Approval
4. TradeConfirm: TradeConfirm

### CODE REVIEW

**Review by reallybeard — APPROVED** [review]

**Comment by 0xdef1cafe** [comment]
@helr441 can you please push an empty commit to trigger ci again with the PR title i updated? :)

`git commit --allow-empty -m "chore: trigger ci" && git push`

**Comment by helr441** [comment]
Seems CI didn't run still. Perhaps you click "Approve and run" under the "1 workflow awaiting approval" part on the PR? https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks.

Since CI didn't run anyway, I'll amend the commits using the project's convention (feat/chore/etc.).

File: src/components/Trade/TradeInput.tsx [hunk 1]
```diff
@@ -163,142 +164,171 @@ export const TradeInput = ({ history }: RouterProps) => {
...
+                    />
+                    <Button
+                      label={'Approval'}
+                      onClick={async () =>
+                        history.push({
+                          pathname: '/trade/approval',
+                          state: { fiatRate: await getGetFiatRate(true) }
+                        })
+                      }
+                    /> */}
```
Comment (cjthompson) [inline src/components/Trade/TradeInput.tsx:289]: I'd prefer not to keep commented out code.

**Review by cjthompson — APPROVED** [review]

File: src/components/Trade/TradeInput.tsx [hunk 2]
```diff
@@ -163,142 +164,171 @@ export const TradeInput = ({ history }: RouterProps) => {
...
+                  <>
+                    <RawText whiteSpace={'pre'}>{`1 ${sellAsset.currency?.symbol} = `}</RawText>
+                    <NumberFormat
+                      value={firstNonZeroDecimal(bnOrZero(quote?.rate))}
+                      displayType={'text'}
+                      thousandSeparator={true}
+                    />
+                    <RawText whiteSpace={'pre'}>{` ${buyAsset?.currency?.symbol}`}</RawText>
+                    <HelperTooltip label={translate('trade.tooltip.rate')} />
+                    {/* <Button
```
Comment (DaoDev44) [inline src/components/Trade/TradeInput.tsx:272]: Can you remove this commented out code?

**Review by DaoDev44 — APPROVED** [review]

**Comment by 0xean** [comment]
@helr441 - if you can remove the commented code, I will get this merged and paid out for you! Thanks for your contribution!

File: src/components/Trade/TradeInput.tsx (hunk 1 above)
Comment (helr441) [reply src/components/Trade/TradeInput.tsx:289]: I will remove this bit and amend the commit.

File: src/components/Trade/TradeInput.tsx (hunk 2 above)
Comment (helr441) [reply src/components/Trade/TradeInput.tsx:272]: I will remove this bit and amend the commit.

**Pull request merged**